geonode/tasks/tasks.py

```diff
@@ -81,3 +82,16 @@ def send_queued_notifications(self, *args):
         send_all(settings.NOTIFICATION_LOCK_LOCATION)
     else:
         send_all(*args)
+
+
+@app.task(bind=True,
+          name='geonode.tasks.layers.set_permissions',
+          queue='default')
+def set_permissions(self, permissions_names, resources_names,
+                    users_usernames, groups_names, delete_flag):
+    from geonode.layers.utils import set_layers_permissions
+    with transaction.atomic():
+        for permissions_name in permissions_names:
+            set_layers_permissions(
+                permissions_name, resources_names, users_usernames, groups_names, delete_flag
+            )
```

Comment:
any exception handling/logging and retry policy (if required) would be nice here